packages/core/utils/lib/convert-query-params.js

```diff
@@ -147,18 +147,72 @@ const convertPopulateQueryParams = (populate, depth = 0) => {
   }
 
   if (_.isPlainObject(populate)) {
-    const transformedPopulate = {};
-    for (const key in populate) {
-      transformedPopulate[key] = convertNestedPopulate(populate[key]);
-    }
-
-    return transformedPopulate;
+    return convertPopulateObject(populate, schema);
   }
 
   throw new InvalidPopulateError();
 };
 
-const convertNestedPopulate = subPopulate => {
+const convertPopulateObject = (populate, schema) => {
+  if (!schema) {
+    return {};
+  }
+
+  const { attributes } = schema;
+
+  return Object.entries(populate).reduce((acc, [key, subPopulate]) => {
+    const attribute = attributes[key];
+
+    if (!attribute) {
+      return acc;
+    }
+
+    // TODO: This is a temporary solution for dynamic zones that should be
+    // fixed when we'll implement a more accurate way to query dynamic zones
+    if (attribute.type === 'dynamiczone') {
+      const generatedFakeDynamicZoneSchema = {
```

Comment:
This is going to add a breaking change in the case two attributes have different types. I really don't think this approach is the way to go here.

Reply:
There was already a collision in the previous way of doing the populate filtering on DZs since we were able to provide any field without having any additional checks.
I don't think it would create a breaking change but I agree it will be equally as problematic (hence the todo notice )